examples/3-adapter_advanced.ipynb

```diff
@@ -0,0 +1,740 @@
+    "                force_fitting=True\n",
+    "                )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ensemble_step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We used a little different syntax in `adapter` this time. Recipe for `y_proba` consists of two things:\n",
+    "- a list of objects returned by input steps that should be used to build `y_proba`,\n",
+    "- a function which merges them into a final `y_proba` object.\n",
+    "\n",
+    "So `[(rf_step.name, 'y_proba'), (xgb_step.name, 'y_proba')]` tells the adapter to extract `y_proba` arrays from dictionaries returned by `rf_step` and `xgb_step` and put them in a list. Then `lambda lst: np.array(lst).mean(axis=0)` will average these arrays.\n",
+    "\n",
+    "An adapter is actually a description of how to build arguments for `fit_transform` and `transform`. It is a dictionary, where:\n",
+    "- keys must agree with transormer's `fit_transform` and `transform` arguments,\n",
+    "- values must be either:\n",
+    "  1. a brick description,\n",
+    "  2. a pair of:\n",
+    "    - a list of brick descriptions,\n",
+    "    - a function that merges extracted results of previous steps,\n",
+    "\n",
+    "where _brick description_ is a pair of node name and key in the dictionary returned by that node.\n",
+    "\n",
+    "Step with an adapter proceeds like this:\n",
+    "1. It gathers results from preceeding nodes.\n",
+    "2. It builds a dictionary with the same keys as the adapter and with values built according to descriptions:\n",
+    "   - if the key in the adapter maps to a single brick description, an appropriate object is extracted from the results of input nodes,\n",
+    "   - in the other case, objects are extracted according to brick descriptions and added to a list, which is then passed to a function that generates final object.\n",
+    "3. Arguments of `fit_transform` and `transform` are filled using the above dictionary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's check if our ensembling works."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ensemble_step.fit_transform(data_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looks fine! However, often we are interested only in the class with the highest probability. Let's make a step that will find this class for us."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "guesses_step = Step(name='guesses_maker',\n",
+    "                 transformer=Dummy(),\n",
+    "                 input_steps=[ensemble_step],                 \n",
+    "                 adapter={'y_pred': ([(ensemble_step.name, 'y_proba')],\n",
```

Comment:
Are we making a step where the adapter does all the computation and then the transformer does nothing? If it's easier to do it this way then we may have to think about re-designing the API...